File: 2870. Minimum Number of Operations to Make Array Empty.java
class Solution {
    public int minOperations(int[] nums) {
      int result = 0;
      Map<Integer, Integer> count = new HashMap<>();
      for(final int num: nums)
      count.merge(num,1,Integer::sum);
      for(final int freq : count.values()){
           if(freq == 1)
           return -1;
           result  += (freq +2) /3;
      }
        return result;
    }
}

File: 2870. Minimum Number of Operations to Make Array Empty.cpp
class Solution {
public:
    int minOperations(vector<int>& nums) {
        unordered_map<int,int> mapping;
        for(int i=0;i<nums.size();i++){
            mapping[nums[i]]++;
        }
        int t_count=0;
        for(auto i:mapping){
            if(i.second==1){
                return -1;
            }
            t_count+=ceil((double)i.second/3);
        }
        return t_count;
    }
};
